packages/jest-resolve/src/__tests__/resolve.test.ts

```diff
@@ -6,16 +6,18 @@
  *
  */
 
-'use strict';
-
 import fs from 'fs';
 import path from 'path';
-import {ModuleMap} from 'jest-haste-map';
-// eslint-disable-next-line import/default
+import HasteMap from 'jest-haste-map';
 import Resolver from '../';
+// @ts-ignore: js file
 import userResolver from '../__mocks__/userResolver';
 import nodeModulesPaths from '../nodeModulesPaths';
 import defaultResolver from '../defaultResolver';
+import {ResolverConfig} from '../types';
+
+// @ts-ignore: types are wrong. not sure how...
```

Comment:
built `haste-map index.d.ts`:
```ts
import HasteFS from './HasteFS';
import HasteModuleMap, { SerializableModuleMap as HasteSerializableModuleMap } from './ModuleMap';
export declare type ModuleMap = HasteModuleMap;
export declare type SerializableModuleMap = HasteSerializableModuleMap;
export declare type FS = HasteFS;
```

totally missing the main export.
Working on how to fix this atm

Reply:
I'm not sure we can fix this without dropping the `export type`s at the top of haste map. Basically, we want both `export = HasteMap` (because making it the default export like it should be would be breaking) and `export type ...`. It would be fine for a normal property (can just put it on the class), but I don't know how we could possibly put types onto `HasteMap`

Reply:
Thanks for digging! I say we keep it as is then, and try to go for a real ESM run at some point in the future.

It's easier with modules like haste map anyways since it's mostly internal and not pluggable like other parts of Jest are

Reply:
Guess the `ModuleMap` and `HasteFS` types are more important anyway, they're the ones used in `Context` etc